xds/internal/resolver/serviceconfig.go

```diff
@@ -185,6 +199,61 @@ func (cs *configSelector) SelectConfig(rpcInfo iresolver.RPCInfo) (*iresolver.RP
 	return config, nil
 }
 
+func (cs *configSelector) generateHash(rpcInfo iresolver.RPCInfo, hashPolicies []*xdsclient.HashPolicy) uint64 {
+	var hash uint64
+	var generatedHash bool
+	for _, policy := range hashPolicies {
+		var policyHash uint64
+		var generatedPolicyHash bool
+		switch policy.HashPolicyType {
+		case xdsclient.HashPolicyTypeHeader:
+			md, ok := metadata.FromIncomingContext(rpcInfo.Context)
+			if !ok {
+				continue
+			}
+			values := md.Get(policy.HeaderName)
+			// If the header isn't present, no-op.
+			if len(values) == 0 {
+				continue
+			}
+			joinedValues := strings.Join(values, ",")
+			if policy.Regex != nil {
```

Comment:
There actually is a bug in the xds_client, where `policy.Regex` is set?

In the current code, `Regex` is never `nil`.
And if `RegexRewrite` is unspecified, it will be a `regex("")`, which will then replace empty strings to `RegexSubstitution` (not exactly sure how Replace() works with `""`)?

Let's fix that.

Reply:
Discussed in chat. Kept this conditional, and also added a conditional in xds about a nil GetRegexRewrite().